catalog/plugin/catalog-plugin-pdf/src/main/java/ddf/catalog/plugin/pdf/thumbnail/PdfThumbnailPlugin.java

```diff
@@ -0,0 +1,57 @@
+package ddf.catalog.plugin.pdf.thumbnail;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.transform.pdf.PdfThumbnailGenerator;
+import ddf.content.operation.UpdateResponse;
+import ddf.content.operation.impl.UpdateResponseImpl;
+import ddf.content.plugin.ContentPlugin;
+import ddf.content.plugin.PluginExecutionException;
+import ddf.content.plugin.PostUpdateStoragePlugin;
+
+public class PdfThumbnailPlugin implements PostUpdateStoragePlugin {
+    private static final Logger LOGGER = LoggerFactory.getLogger(PdfThumbnailPlugin.class);
+
+    private final PdfThumbnailGenerator pdfThumbnailGenerator;
+
+    public PdfThumbnailPlugin(PdfThumbnailGenerator pdfThumbnailGenerator) {
+
+        this.pdfThumbnailGenerator = pdfThumbnailGenerator;
+    }
+
+    @Override
+    public UpdateResponse process(UpdateResponse input) throws PluginExecutionException {
+        byte[] thumbnailBytes = null;
+
+        try (InputStream contentStream = input.getUpdatedContentItem().getInputStream()) {
+            thumbnailBytes = pdfThumbnailGenerator.generatePdfThumbnail(contentStream);
```

Comment:
Right now i only generate a thumbnail for the first page (at a configurable DPI).